packages/mdc-toolbar/foundation.js

```diff
@@ -122,6 +121,7 @@ export default class MDCToolbarFoundation extends MDCFoundation {
     const newToolbarRowHeight = this.getRowHeight_();
     if (newToolbarRowHeight !== this.calculations_.toolbarRowHeight) {
       this.calculations_.toolbarRowHeight = newToolbarRowHeight;
+      this.initKeyRatio_();
```

Comment:
I am afraid the logic is wrong here.

The `initKeyRatio_` should be calculate only once at initialization time. In this implementation, the key ratios will be re-calculated every time when user resize the window. First, it is not necessary because key ratio is not related to screen width and more importantly, if the user resize the window when toolbar is in the middle of transition (for example, 50% of the flexible space shrinks), this re-calculation will make the behavior completely off.

However, I think this is one good case that is not currently captured by our test case. I think it worth us to re-think how we can test it more rigorously.